vdev/src/commands/package/msi.rs

```diff
@@ -0,0 +1,149 @@
+            let msi_x64_dir = Path::new("target").join("msi-x64");
+            fs::remove_dir_all(&msi_x64_dir).ok();
+            fs::create_dir_all(&msi_x64_dir)?;
+            fs::copy("distribution/msi", &msi_x64_dir)?;
+
+
+            let artifacts_dir = Path::new("target").join("artifacts");
+            let zip_file = format!("vector-{archive_version}-x86_64-pc-windows-msvc.zip");
+            fs::copy(artifacts_dir.join(&zip_file), msi_x64_dir.join(&zip_file))?;
+
+            // Ensure we are in the `msi-x64` directory
+            env::set_current_dir(&msi_x64_dir)?;
+
+            // Extract the zip file with PowerShell and build the MSI package
+            let powershell_command = format!(
+                "$progressPreference = 'silentlyContinue'; Expand-Archive {zip_file}"
+            );
+            app::exec("powershell", ["-Command", &powershell_command], false)?;
+            build(&archive_version, &package_version)?;
+
+            // Change the current directory back to the original path
+            env::set_current_dir(app::path())?;
+
+            // Copy the MSI file to the artifacts directory
+            let msi_file = format!("vector-{archive_version}-x64.msi");
+            let dest_file = artifacts_dir.join(msi_file);
+            fs::copy(msi_x64_dir.join("vector.msi"), dest_file)?;
+        }
+        Ok(())
+    }
+}
+
+#[cfg(windows)]
+fn build(archive_version: &str, package_version: &str) -> Result<()> {
+    println!("Running Build with args: {archive_version}");
+    println!("Copying ZIP archive...");
+
+    println!("Preparing LICENSE.rtf..");
+    let mut license_rtf_file = File::create("LICENSE.rtf")?;
+    writeln!(
+        license_rtf_file,
+        "{{\\rtf1\\ansi\\ansicpg1252\\deff0\\nouicompat{{\\fonttbl{{\\f0\\fnil\\fcharset0 Lucida Console;}}}}\n\\viewkind4\\uc1\n\\pard\\f0\\fs14\\lang1033\\par"
+    )?;
+
+    let license_content_path = format!("vector-{archive_version}-x86_64-pc-windows-msvc/LICENSE.txt");
+    let license_content = std::fs::read_to_string(license_content_path)?;
+    for line in license_content.lines() {
+        writeln!(license_rtf_file, "{line}\\")?;
+    }
+    writeln!(license_rtf_file, "\n}}")?;
+
+    println!("Substituting version...");
+    let vector_tmpl = std::fs::read_to_string("vector.wxs.tmpl")?;
+    let vector_tmpl_updated = vector_tmpl.replace("${VERSION}", package_version);
+    let mut vector_wxs_file = File::create("vector.wxs")?;
+    writeln!(vector_wxs_file, "{vector_tmpl_updated}")?;
+
+    println!("Building the MSI package...");
+    let vector_dir = format!("vector-{archive_version}-x86_64-pc-windows-msvc");
+    let args = &[
+        &format!("dir {vector_dir}"),
+        "-cg Vector",
+        "-dr INSTALLDIR",
+        "-gg",
+        "-sfrag",
+        "-srd",
+        "-var var.VectorDir",
+        "-out components.wxs"
+    ];
+    Command::new("heat").args(args).capture_output()?;
```

Comment:
Since we don't use the output, and any messages from this command (and below) would end up being thrown away by this, I think you want `check_run` here:
```suggestion
    Command::new("heat").args(args).check_run()?;
```

Reply:
Implemented locally and will push